### PR TITLE
[plugin.video.catchuptvandmore] 0.2.25

### PR DESCRIPTION
**Pull request by sy6sy2**
### Description

Hi @enen92,

Sorry for this "big diff" 😕. In this release we take advantage of the new feature available in Codequick 0.9.13 so we don't need our "bridge" functions anymore in `main.py` and we can directly specify the real callback function path of each channel in the skeleton files. This why a lot a files have been modified.

On top of that, as usual, this release also brings some fixes for channels and new contents as you can see in the changelog.

Thank you for your work 🍺 

cc @wwark, @jmzambon

**Version 0.2.25 changelog:**

> [FR - MYTF1] Move all TF1 group channels in MYTF1 folder
> [FR - MYTF1] Add search feature
> [FR - 6play] Move all M6 group channels in 6play folder
> [FR - myCANAL] Move all Canal+ group channels in myCANAL folder
> [FR - NRJ Play] Move NRJ 12 and Chérie 25 in NRJ Play folder
> [FR - FR3 Régions] Fix typo Franche-Comté
> [BE - RTLplay] Move all RTL group channels in RTLplay folder
> [BE - RTBF] Fix Live TV with no audio (DRM)
> [CH - RTS] Fix Live TV with no audio (DRM)
> [Core] Bump CodeQuick 0.9.13 - Kill xxxx_bridge functions
> [NL - NPO] Fix Lives TV (Audio drop, NPO 1 and 2, Rename NPO 3 to NPO Zapp)
> [FR - Equidia] Add Catchup-TV (Courses and Emissions)
> [CA - NOOVO.TV] Fix Catchup-TV
> [UK - STV] Fix catch up entry error
> [NL - NPO] Fix Next Page error
> [FR - B Smart] Add Catchup-TV
> [FR - Sport En France] Fix Live TV

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py [hunk 1]
```diff
@@ -31,23 +31,137 @@
...
-# Add Replay
 
 URL_ROOT = "https://www.bsmart.fr"
 
 URL_LIVE_DATAS = URL_ROOT + "/static/js/bundle.%s.js"
 
+URL_API = 'https://api.bsmart.fr/api'
+
+CATEGORIES_BSMART = {
+    '/programs/': 'Emissions',
```
Comment (learningit) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py]: Shouldn't these be in strings.po?

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/equidia.py [hunk 2]
```diff
@@ -29,25 +29,168 @@
...
+URL_API_SEARCH = "https://api.equidia.fr/api/public"
+
+URL_IMAGE = URL_API_SEARCH + '/media/article_header/%s'
+# ImageId
+
+URL_REPLAY_DATAS = 'https://api.equidia.fr/api/public/videos-store/player/%s'
+# VideoId
+
+CATEGORIES_VIDEOS_EQUIDIA = {
+    '/search/emissions': 'Emissions',
```
Comment (learningit) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/equidia.py]: Shouldn't these be in strings.po also?

**Comment by learningit** [comment]
I only reviewed the diff. It looks good except for the minor strings.po issues below.

One note: The CBS News channel you have received a take down request from CBS to Kodi.org quite a bit ago. I know because I had an addon for it. Not sure what that means for other add-ons in the kodi.org repo.

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py (hunk 1 above)
Comment (enen92) [reply plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py]: yeah those need to be localized @sy6sy2

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py [hunk 3]
```diff
@@ -107,11 +108,28 @@
...
+
+    # (item_id, label, thumb, fanart)
+    channels = [
+        ('rtl_tvi', 'RTL TVI', 'rtltvi.png', 'rtltvi_fanart.jpg'),
+        ('club_rtl', 'CLUB RTL', 'clubrtl.png', 'clubrtl_fanart.jpg'),
+        ('plug_rtl', 'PLUG RTL', 'plugrtl.png', 'plugrtl_fanart.jpg'),
+        ('rtl_info', 'RTL INFO', 'rtlinfo.png', 'rtlinfo_fanart.jpg'),
+        ('rtl_sport', 'RTL Sport', 'rtlsport.png', 'rtlsport_fanart.jpg'),
+        ('bel_rtl', 'BEL RTL', 'belrtl.png', 'belrtl_fanart.jpg'),
+        ('contact', 'Contact', 'contact.png', 'contact_fanart.jpg')
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py:122]: is contact a channel or a way of contact? if the later where does the message go to?

**Review by enen92 — COMMENTED** [review]
Only the localization stuff needed and the contact question answered although I think it is just a channel (haven't seen anything on the code that could tell me otherwise)

File: plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py (hunk 3 above)
Comment (sy6sy2) [reply plugin.video.catchuptvandmore/resources/lib/channels/be/rtlplaybe.py:122]: Yes, Contact is a Belgium radio from the RTL group (https://www.rtlplay.be/contact). Good Radio by the way 😛

**Comment by sy6sy2** [comment]
Should be ok now, thank you for the review!

**Review by enen92 — APPROVED** [review]

**Pull request merged**